src/nemo_evaluator/engine/model_client.py

```
@@ -354,8 +362,16 @@ async def _post_with_retry(self, url: str, payload: dict[str, Any]) -> dict:
                             continue
 
                         resp.raise_for_status()
-                        return await resp.json()
+                        data = await resp.json()
+                        if not isinstance(data, dict):
+                            raise InfraError(
+                                f"Model returned non-object JSON response ({type(data).__name__}); "
+                                "expected a JSON object."
+                            )
```

Ruff (0.15.21)

[warning] Abstract `raise` to an inner function

(TRY301)